charts/kueue/templates/rbac/role.yaml

```diff
@@ -82,6 +82,7 @@ rules:
   - apiGroups:
       - apps
     resources:
+      - deployments
```

Comment:
@mbobrovskyi can you please explain in more detail in the PR description why this is needed? It does not show up in the issue description (only AppWrapper , JobSet and Job) so, I'm surprised seeing this, but haven't read the PR in detail yet.

Reply:
Yeah, but we supported Deployments in the past without it, so I'm wondering how it worked - or maybe it didn't work and this is yet another issue?

Reply:
Also, if this is a bugfix, it would be good to have a test which fails otherwise - do we have such an e2e test?

Reply:
Deployment doesn’t have a workload, so previously it was just ignored at the ReplicaSet level. Since ReplicaSet is not an integration, the Pod is treated as the top-level job. Now we traverse to top level job event if it is not integration.

Reply:
After last changes we don't need it. Due to we are skipping suspend by parent logic for Pod integration at all.